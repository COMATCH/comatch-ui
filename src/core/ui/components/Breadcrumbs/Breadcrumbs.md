## Breadcrumbs

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**className** | `String` | `null` | :x: | 
**history** | `Array[]<String>` | `[]` | :x: | 
**id** | `String` | `null` | :x: | 

