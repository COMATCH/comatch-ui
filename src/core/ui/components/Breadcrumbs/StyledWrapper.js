import styled from 'styled-components';

const StyledWrapper = styled.section`

ul {
    list-style: none;
    display: flex;
    flex-direction: row;
}

li {
    font-size: 12px;
    color: #777777;
    margin: 3px;
}

li:last-child {
    font-weight: bold;
}

`;

export { StyledWrapper };



