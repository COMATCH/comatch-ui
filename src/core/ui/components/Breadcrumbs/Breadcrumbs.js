import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { StyledItem, StyledWrapper } from './StyledWrapper';

const propTypes = {
    id: PropTypes.string,
    className: PropTypes.string,
    history: PropTypes.arrayOf(PropTypes.string),
};

const defaultProps = {
    id: null,
    className: null,
    history: [],
};

export const Breadcrumbs = ({ id, className, history }) => (
    <StyledWrapper {...(!!id && { id })} className={classnames('Breadcrumbs', className)}>
        {history.map((route, index) => (
            <StyledItem className='list-item' key={route}>{`${index ? '> ' : ''}${route}`}</StyledItem>
        ))}
    </StyledWrapper>
);

Breadcrumbs.propTypes = propTypes;
Breadcrumbs.defaultProps = defaultProps;
