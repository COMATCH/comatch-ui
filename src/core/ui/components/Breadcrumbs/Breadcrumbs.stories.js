import React from 'react';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { Breadcrumbs } from './Breadcrumbs';

const items = ['Item 1', 'Item 2', 'Item 3'];

storiesOf('Breadcrumbs', module)
    .add('with basic style', () => (
        <>
            <Breadcrumbs history={items} />
            <Highlight className="html">{'<Breadcrumbs history={["Item 1", "Item 2", "Item 3"]} />'}</Highlight>
        </>
    ))
    .add('with class & ID', () => (
        <>
            <Breadcrumbs id="ID" className="IDK" history={items} />
            <Highlight className="html">
                {'<Breadcrumbs id="ID" className="IDK" history={["Item 1", "Item 2", "Item 3"]} />'}
            </Highlight>
        </>
    ));
