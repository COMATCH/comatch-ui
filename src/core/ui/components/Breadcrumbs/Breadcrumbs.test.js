import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Breadcrumbs } from './Breadcrumbs';
import '../../../config/tests/setup';

describe('Breadcrumbs', () => {
    it('should render breadcrumbs correctly', () => {
        const breadcrumbs = renderer.create(<Breadcrumbs history={['Item 1', 'Item 2']} />).toJSON();
        expect(breadcrumbs).toMatchSnapshot();
    });

    it('should have the correct default props', () => {
        const breadcrumbs = mount(<Breadcrumbs />);
        expect(breadcrumbs.prop('id')).toEqual(null);
        expect(breadcrumbs.prop('className')).toEqual(null);
        expect(breadcrumbs.prop('history')).toEqual([]);
    });

    it('should render passed props correctly', () => {
        const items = ['Item 1', 'Item 2', 'Item 3', 'Item 4', 'Item 5', 'Item 6'];
        const breadcrumbs = mount(<Breadcrumbs id="ID" className="IDK" history={items}></Breadcrumbs>);

        expect(breadcrumbs.prop('id')).toEqual('ID');
        expect(breadcrumbs.prop('className')).toEqual('IDK');
        expect(breadcrumbs.prop('history')).toEqual(items);
    });
});
