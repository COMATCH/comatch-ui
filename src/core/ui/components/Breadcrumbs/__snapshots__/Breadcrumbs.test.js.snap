// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should render breadcrumbs correctly 1`] = `
<section
  className="Breadcrumbs sc-bdVaJa fiviaV"
>
  <ul>
    <li>
      Item 1
    </li>
    <li>
      &gt;
    </li>
    <li>
      Item 2
    </li>
  </ul>
</section>
`;
